### PR TITLE
fix: upgrade to pdfjs v4

### PR DESCRIPTION
**Pull request by jhpedemonte**
#### What do these changes do/fix?

Upgrade to latest PDF.js build.

Implements https://github.ibm.com/Watson-Discovery/disco-issue-tracker/issues/19108

#### Are there any breaking changes included in this pull request?

- Check PDF views in Storybook
- Check PDF view in example app. To see a PDF in example app, create a project and upload "Art Effects.pdf". In app, search for "art effects" and pick first result to see PDF view.

### CODE REVIEW

**Comment by jhpedemonte** [comment]
### To do
- [ ] Fix tests
- [ ] Clean up `packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/typings.d.ts`

**Comment by noah-eigenfeld** [comment]
Screenshot 2024-07-31 at 11 58 35 AM
Saw this on my own branch when I last tried to make updates, and I still see it now on yours: for some reason the height of the rendered textLayer is twice the required height of the PDF itself. Nothing breaks because of this (at least, I don't think it does), but the user can scroll really far down in the editor.

File: packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/utils.test.tsx [hunk 1]
```diff
@@ -8,19 +8,34 @@ describe('utils', () => {
...
       const arrayBufferViewData: ArrayBufferView = new DataView(new ArrayBuffer(2));
-      expect(toPDFSource(arrayBufferViewData as TypedArray).data).toBe(arrayBufferViewData);
+      const pdfSource = toPDFSource(arrayBufferViewData as TypedArray);
+      // convert to a typed array to verify underlying data is the same
+      expect(
+        areTypeArraysEqual(
+          new Uint8Array((pdfSource.data as unknown as DataView).buffer),
+          new Uint8Array(arrayBufferViewData.buffer)
+        )
+      ).toBe(true);
```
Comment (noah-eigenfeld) [inline packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/__tests__/utils.test.tsx:38]: Thanks for fleshing out these tests more

**Review by noah-eigenfeld — COMMENTED** [review]
Code changes look good to me. Waiting on an approval until some bugs observed in testing are fixed

**Comment by jhpedemonte** [comment]
> the height of the rendered textLayer is twice the required height of the PDF itself

You're still seeing this? I explicitly updated the CSS to fix this, so odd that it would still be around.

**Review by noah-eigenfeld — APPROVED** [review]

**Pull request merged**